docs/generator.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"text/template"
+	"github.com/resumic/schema/docs/util"
+	"github.com/tidwall/gjson"
+)
+
+func main() {
+	var conf, doc, tmpl string
+	flag.StringVar(&conf, "conf", "schema.json", "Schema file")
+	flag.StringVar(&tmpl, "tmpl", "docs/dev/doc.tmpl", "Template file")
+	flag.StringVar(&doc, "doc", "docs/dev/doc.md", "Output file")
+	content, err := ioutil.ReadFile(conf)
+	if err != nil {
+		log.Fatal(err)
+	}
+	f, err := os.OpenFile(tmpl, os.O_APPEND, 0666)
+	value := gjson.GetBytes(content, "properties")
+	value.ForEach(func(key, value gjson.Result) bool {
+		_, err = f.WriteString("\n# " + strings.Title(key.String()) + "\n" + "This section is of " + "{{.Properties." + strings.Title(key.String()) + ".Type}} type that tells about the basic information of the user and consists of the following sub-sections:\n") //print name of sections
+		if err != nil {
+			log.Fatal(err)
+		}
+		println(key.String())
+		result := gjson.GetBytes(content, "properties."+key.String())
+		result.ForEach(func(key1, value gjson.Result) bool {
+			_, err = f.WriteString("##    " + key1.String() + "" + "\n") //print name of sub-sections
```

Comment:
For some reason GitHub marks my comments as resolved so I'm writing them again:

In addition to whitespace, there's no need for the `+ "" +`, this is the same as just writing `+`

Reply:
@alarsyo Thanks for reminding, forgot to remove this while editing the template.